pkg/execution/state/driver_response_test.go

```diff
@@ -218,6 +218,63 @@ func TestDriverResponseFinal(t *testing.T) {
 	}
 }
 
+func TestDriverResponseUserError(t *testing.T) {
+	tests := []struct {
+		name     string
+		r        DriverResponse
+		expected map[string]any
+	}{
+		{
+			name: "with no Output and Err",
+			r:    DriverResponse{Output: nil, Err: fmt.Errorf("something went wrong")},
+			expected: map[string]any{
+				"error": fmt.Errorf("something went wrong"),
+			},
+		},
+		{
+			name: "with no Output and no Err",
+			r:    DriverResponse{Output: nil},
+			expected: map[string]any{
+				"error": "Unknown error running SDK",
```

Comment:
Is there ever a case where `Output` and `Err` will both be null?

Reply:
Yes, but in that case we won't have an error and this shouldn't hit!